src/snowflake/snowpark/modin/pandas/__init__.py

```diff
@@ -346,3 +346,7 @@ def __getattr__(name: str) -> Any:
 modin.core.dataframe.algebra.default2pandas.default.DefaultMethod.register = (
     snowflake.snowpark.modin.core.dataframe.algebra.default2pandas.default.DefaultMethod.register
 )
+
+# Call getOrCreate() at import time on behalf of user so that in a stored procedure,
+# the user does not need to do it explicitly.
+snowflake.snowpark.Session.SessionBuilder().getOrCreate()
```

Comment:
@sfc-gh-helmeleegy for snowpark pandas, we by design to ask the user to explicitly set up the session to stay consistent with sowpark usage, and we do not require customer to provide connection files , so getOrCreate() could fail if user does not have a connection file. 

With stored sproc, i thought the session object is automatically created, and snowaprk pandas have the capablity to get the current active session for usage.  For some reason, that seems not the case. @sfc-gh-sfan why it is happening in such way?

Reply:
actually @sfc-gh-sfan I have a question, do you have a code pointer on how the session creation is handled in stored sproc today?

Reply:
After making the session argument optional in the proc's handler, the session object was not being created anymore in the case when the session argument was not specified. That's because it was not being passed along with the other arguments to the handler anymore.

@sfc-gh-zyao has a [draft PR](https://github.com/snowflakedb/snowflake/pull/193694) for pre-creating session objects even for session-less handlers, which seems promising. Until this becomes available, we will need to either pass the session argument, or create it at the sproc side using `getOrCreate`. Currently, it's left to the user to create it inside the body of the handler.

This PR attempts to automatically create the session object at package import time (only for sprocs) so that users do not need to worry about creating it themselves.

Reply:
Yeah, i think the direction @sfc-gh-zyao is the right direction. Our original intention is to make the argument optional, not making the session object creation optional. I was talking to @sfc-gh-zyao , he mentioned he will work on this to get the fix in soon. I think we can just wait for this fix, instead of adding the workaround here, since it also took long time for the package to be released in stored sproc also.